server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -267,6 +297,82 @@ public void restoreSnapshot(final RestoreRequest request, final ActionListener<R
         });
     }
 
+    private boolean allTemplates(List<String> resolvedTemplates) {
+        return resolvedTemplates.size() == 1 && resolvedTemplates.get(0).equals("_all");
+    }
```

Comment:
Since this code is used in Logical Replication as well, we cannot remove `request.template()/indices()`, at least right now without refactoring Logical replication to operate with CrateDB specific components.

Thus, I created resovedTemplates/Indices which are taken from request in case of LR usage in order not deal with version check/List<RelationName> != null check everywhere we access indices or templates. 

hence, all request.indices() and request.templates() usages are gone